fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2110,36 +2064,29 @@ public Range getUnbuiltRange() {
-                                                                                         @Nonnull StoreExistenceCheck existenceCheck,
-                                                                                         @Nonnull IsolationLevel storeHeaderIsolationLevel,
-                                                                                         @Nonnull IsolationLevel indexStateIsolationLevel) {
-        // TODO: Can become non-static once the static variants of loadRecordStoreStateAsync are removed
-        CompletableFuture<RecordMetaDataProto.DataStoreInfo> storeHeaderFuture = FDBRecordStore.loadStoreHeaderAsync(context, subspaceProvider, subspace, existenceCheck, storeHeaderIsolationLevel);
-        CompletableFuture<Map<String, IndexState>> loadIndexStates = loadIndexStatesAsync(context, subspace, indexStateIsolationLevel);
+    private CompletableFuture<RecordStoreState> loadRecordStoreStateInternalAsync(@Nonnull StoreExistenceCheck existenceCheck,
+                                                                                  @Nonnull IsolationLevel storeHeaderIsolationLevel,
+                                                                                  @Nonnull IsolationLevel indexStateIsolationLevel) {
+        CompletableFuture<RecordMetaDataProto.DataStoreInfo> storeHeaderFuture = loadStoreHeaderAsync(existenceCheck, storeHeaderIsolationLevel);
+        CompletableFuture<Map<String, IndexState>> loadIndexStates = loadIndexStatesAsync(indexStateIsolationLevel);
         return context.instrument(FDBStoreTimer.Events.LOAD_RECORD_STORE_STATE, storeHeaderFuture.thenCombine(loadIndexStates, RecordStoreState::new));
     }
 
     @Nonnull
-    private static CompletableFuture<Map<String, IndexState>> loadIndexStatesAsync(@Nonnull FDBRecordContext context,
-                                                                                   @Nonnull Subspace subspace,
-                                                                                   @Nonnull IsolationLevel isolationLevel) {
-        // TODO: Can become non-static once the static variants of loadRecordStoreStateAsync are removed
-        Subspace isSubspace = subspace.subspace(Tuple.from(INDEX_STATE_SPACE_KEY));
+    private CompletableFuture<Map<String, IndexState>> loadIndexStatesAsync(@Nonnull IsolationLevel isolationLevel) {
+        Subspace isSubspace = getSubspace().subspace(Tuple.from(INDEX_STATE_SPACE_KEY));
         KeyValueCursor cursor = KeyValueCursor.Builder.withSubspace(isSubspace)
-                .setContext(context)
+                .setContext(getContext())
                 .setRange(TupleRange.ALL)
                 .setContinuation(null)
                 .setScanProperties(new ScanProperties(ExecuteProperties.newBuilder().setIsolationLevel(isolationLevel).build()))
                 .build();
-        FDBStoreTimer timer = context.getTimer();
+        FDBStoreTimer timer = getContext().getTimer();
```

Comment:
```suggestion
        FDBStoreTimer timer = getTimer();
```

Reply:
At the moment, I think that `store.getTimer()` is actually implemented by just asking the context for its timer, but I think this protects us if we ever make that more sophisticated.

Reply:
Sure. Thanks for the review.